reconcile/openshift_users.py

```diff
@@ -1,34 +1,56 @@
-import logging
 import itertools
+import logging
+import sys
 
 from sretoolbox.utils import threaded
 
-from reconcile import openshift_groups
-from reconcile import openshift_rolebindings
-from reconcile import queries
-
-from reconcile.utils.oc import OC_Map
+from reconcile import (
+    openshift_groups,
+    openshift_rolebindings,
+    queries,
+)
 from reconcile.utils.defer import defer
-
+from reconcile.utils.oc import OC_Map
 
 QONTRACT_INTEGRATION = "openshift-users"
 
 
-def get_cluster_users(cluster, oc_map):
+def get_cluster_users(cluster, oc_map, clusters):
     oc = oc_map.get(cluster)
     if not oc:
         logging.log(level=oc.log_level, msg=oc.message)
         return []
-    users = [
-        u["metadata"]["name"]
-        for u in oc.get_users()
-        if u.get("identities", None) is not None
-        and len(u["identities"]) == 1
-        and u["identities"][0].startswith("github")
-        and not u["metadata"].get("labels", {}).get("admin", "")
-    ]
+    users: list[str] = []
 
-    return [{"cluster": cluster, "user": user} for user in users or []]
+    cluster_info = None
+    for cl in clusters:
+        if cl["name"] == cluster:
+            cluster_info = cl
```

Comment:
nit: I like this fancy way of finding an item from a list using a generator expression. This is very subjective so feel free to just close this :)
```suggestion
cluster_info = next((cl for cl in clusters if cl["name"] == cluster), None)
```

Reply:
I like your way to find an item in a list :) Thx for that